src/ui/components/Notice/styles.scss

```diff
@@ -199,6 +199,43 @@
   }
 }
 
+.Notice-warning {
+  .Notice-button {
+    @include button(
+      $background: $yellow-60,
+      $background-active: $yellow-80,
+      $background-hover: $yellow-70,
+      $color: $yellow-90
+    );
+
+    &:active,
+    &:hover,
+    &:focus {
+      color: $white;
+    }
+
+    @include notice_button();
+  }
+
+  .Notice-icon {
+    background-image: url('./img/warning.svg');
+    background-size: cover;
+  }
+
+  .IconXMark-path {
+    fill: $yellow-90;
+  }
+
+  background-color: $yellow-50;
+
+  &,
+  :link,
+  :hover,
+  :active {
+    color: $yellow-90;
+  }
+}
```

Comment:
I suppose you copy/pasted the whole SCSS declaration for a "valid type"? Looks like a mixin could be extracted..

Reply:
I'll give that a try.

Reply:
I started looking at this, and it doesn't look like we can use substitution inside variable names, not without defining a lot of extra things to support it anyway. For example, what I think we would want to do here is have a mixin that accepts a `$color-name` argument and then use that to define all the colors. So, for example, instead of an explicit:

```scss
    @include button(
      $background: $yellow-60,
      $background-active: $yellow-80,
      $background-hover: $yellow-70,
      $color: $yellow-90
    );
```

We could have something where `yellow` above would be substituted with `$color-name`. I couldn't seem to get that to work though. Without that, there doesn't seem to be a lot of value in introducing a mixin, as we'll still need specific code for most of the classes.

@willdurand Is there a trick to doing that substitution that you know of? If not, do you agree that a mixin would be of little value?

Reply:
mmh, I don't think we can create a variable based on another variable value indeed.

The mixin could take 4 or 5 variables for the different colors and one for the backgrounnd image, and that would still reduce the amount of repeated code.

Reply:
Ok, I took this approach. There are a lot of variables, but it does still reduce the overall amount of code. Thanks!